clarity_ext/utility/testing_parse_scripts/builders.py

```diff
@@ -0,0 +1,122 @@
+from mock import MagicMock
+from StringIO import StringIO
+from io import BytesIO
+from clarity_ext.service.file_service import Csv
+from clarity_ext.service.file_service import FileService
+from clarity_ext.service.process_service import ProcessService
+from clarity_ext.context import ExtensionContext
+from clarity_ext.service.artifact_service import ArtifactService
+from clarity_ext.domain.process import Process
+from clarity_ext.domain.user import User
+from clarity_ext.domain.udf import UdfMapping
+from clarity_ext.utility.testing_parse_scripts.fake_artifact_factory import FakeArtifactFactory
+
+
+class ContextBuilder:
+    """
+    Add entities to repositories held by context
+    E.g. shared files, artifacts, analytes.
+    """
+    def __init__(self):
+        self.step_repo = FakeStepRepo()
+        self.logger = FakeLogger()
+        self.context = None
+        self._create()
+
+    def with_analyte_pair(self, input, output):
+        self.step_repo.add_analyte_pair(input, output)
+
+    def with_mocked_local_shared_file(self, filename, contents):
+        monkey = LocalSharedFilePatcher()
+        monkey.cache[filename] = contents
+        self.context.local_shared_file = monkey.local_shared_file
+
+    def _create(self):
+        session = MagicMock()
+        file_repository = MagicMock()
+        artifact_service = ArtifactService(self.step_repo)
+        os_service = MagicMock()
+        current_user = None
+        file_service = FileService(
```

Comment:
There seems to be a mix of Mocks and actual implementations here, e.g. `FileService` and `ProcessService` use the actual implementations. Is there any particular reason for this?

Reply:
Cool. I think that makes sense, and avoids the problem I was trying to adress. :+1: